src/Reports.tsx

```diff
@@ -45,39 +52,37 @@ export const Reports: React.FC = () => {
         let payload = n?.payload;
         if (payload) {
             try {
-                payload = ethers.utils.toUtf8String(payload);
+                payload = fromHex(payload as `0x${string}`, 'string');
             } catch (e) {
                 payload = payload + " (hex)";
             }
         } else {
             payload = "(empty)";
         }
         return {
-            id: `${n?.id}`,
-            index: parseInt(n?.index),
+            index: n.index,
             payload: `${payload}`,
-            input: n ? {index:n.input.index,payload: inputPayload} : {},
+            input: (n && n.input?.id) ? {id:n.input.id,payload: inputPayload} : undefined,
         };
     }).sort((b: any, a: any) => {
         if (a.input.index === b.input.index) {
             return b.index - a.index;
         } else {
-            return b.input.index - a.input.index;
+            return Number(BigInt(b.input.id) - BigInt(a.input.id));
```

Comment:
It's better to sort by report index, right?

Reply:
Indeed
This is a legacy ordering that the report index was local to the input